.github/workflows/contracts-ecdsa.yml

```
@@ -298,3 +315,90 @@ jobs:
         run: |
           yarn run hardhat --network ${{ github.event.inputs.environment }} \
             etherscan-verify
+
+  # This job is responsible for publishing packackes with slightly modified
+  # contracts. The modifications are there to help with the process of testing
+  # some features on the T Token Dashboard. The job starts only if workflow
+  # gets triggered by the `workflow_dispatch` event on the branch called
+  # `dapp-development`.
+  contracts-dapp-development-deployment-testnet:
+    needs: [contracts-build-and-test]
+    # TODO: Remove ` || github.ref == 'refs/heads/ci-dapp-dev-test'` before merge to main
+    if: |
+      github.event_name == 'workflow_dispatch'
+        && (github.ref == 'refs/heads/dapp-development' || github.ref == 'refs/heads/ci-dapp-dev-test')
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./solidity/ecdsa
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "14.x"
+          cache: "yarn"
+          cache-dependency-path: solidity/ecdsa/yarn.lock
+          registry-url: "https://registry.npmjs.org"
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Get upstream packages versions
+        uses: keep-network/ci/actions/upstream-builds-query@v2
+        id: upstream-builds-query
+        with:
+          upstream-builds: ${{ github.event.inputs.upstream_builds }}
+          query: |
+            solidity-contracts-version = github.com/threshold-network/solidity-contracts#version
+            random-beacon-version = github.com/keep-network/keep-core/solidity/random-beacon#version
+
+      - name: Resolve latest contracts
+        run: |
+          yarn upgrade \
+            @threshold-network/solidity-contracts@${{ steps.upstream-builds-query.outputs.solidity-contracts-version }} \
```

Non-blocking suggestion
```suggestion
            threshold-contracts-version = github.com/threshold-network/solidity-contracts#version
            random-beacon-version = github.com/keep-network/keep-core/solidity/random-beacon#version

      - name: Resolve latest contracts
        run: |
          yarn upgrade \
            @threshold-network/solidity-contracts@${{ steps.upstream-builds-query.outputs.threshold-contracts-version }} \
```

---

[c6556fe](https://github.com/keep-network/keep-core/pull/3121/commits/c6556fe97d5a1d37b52e11fad42bffbb403d5bc5)